src/components/film/film.service.ts

```diff
@@ -25,111 +22,63 @@ import { FilmRepository } from './film.repository';
-        session,
-      );
-
-      this.logger.debug(`flim created`, { id: result.id });
-      return await this.readOne(result.id, session);
-    } catch (exception) {
-      this.logger.error(`Could not create film`, {
-        exception,
-        userId: session.userId,
-      });
-      throw new ServerException('Could not create film', exception);
-    }
+    const dto = await this.repo.create(input, session);
+    this.privileges.for(session, Film, dto).verifyCan('create');
+    return this.secure(dto, session);
   }
 
   @HandleIdLookup(Film)
   async readOne(id: ID, session: Session, _view?: ObjectView): Promise<Film> {
-    this.logger.debug(`Read film`, {
-      id,
-      userId: session.userId,
-    });
-
     const result = await this.repo.readOne(id);
-    return await this.secure(result, session);
+    return this.secure(result, session);
   }
 
   async readMany(ids: readonly ID[], session: Session) {
     const films = await this.repo.readMany(ids);
-    return await Promise.all(films.map((dto) => this.secure(dto, session)));
+    return films.map((dto) => this.secure(dto, session));
   }
 
-  private async secure(dto: DbTypeOf<Film>, session: Session): Promise<Film> {
-    return this.privileges.for(session, Film).secure({
-      ...dto,
-      scriptureReferences: this.scriptureRefs.parseList(
-        dto.scriptureReferences,
-      ),
-    });
+  private secure(dto: UnsecuredDto<Film>, session: Session): Film {
+    return this.privileges.for(session, Film).secure(dto);
   }
 
   async update(input: UpdateFilm, session: Session): Promise<Film> {
-    const film = await this.readOne(input.id, session);
+    const film = await this.repo.readOne(input.id);
     const changes = {
       ...this.repo.getActualChanges(film, input),
       scriptureReferences: ifDiff(isScriptureEqual)(
         input.scriptureReferences,
-        film.scriptureReferences.value,
+        film.scriptureReferences,
       ),
     };
     this.privileges.for(session, Film, film).verifyChanges(changes);
-    const { scriptureReferences, ...simpleChanges } = changes;
-
-    await this.scriptureRefs.update(input.id, scriptureReferences);
 
-    await this.repo.update(film, simpleChanges);
-
-    return await this.readOne(input.id, session);
+    const updated = await this.repo.update({ id: input.id, ...changes });
+    return this.secure(updated, session);
   }
 
   async delete(id: ID, session: Session): Promise<void> {
-    const film = await this.readOne(id, session);
+    const film = await this.repo.readOne(id);
 
     this.privileges.for(session, Film, film).verifyCan('delete');
 
     try {
       await this.repo.deleteNode(film);
     } catch (exception) {
-      this.logger.error('Failed to delete', { id, exception });
       throw new ServerException('Failed to delete', exception);
     }
-
-    this.logger.debug(`deleted film with id`, { id });
   }
 
   async list(input: FilmListInput, session: Session): Promise<FilmListOutput> {
```

Comment:
@CarsonF Just saw that a missed fixing the signature here.  Will get that fixed and pushed